### ISSUE TITLE
Mobile performance issues

### ISSUE THREAD

**Issue by jrosindell** [OneZoom/OZtree#85]

**Comment by jrosindell** [comment]
1.) Scrolling of elements in UI layer doesn't permit us to see the top of the elements - hidden by URL bar

2.) (nice if possible) unwanted scrolling and pinging of background to page

3.) the entire page is being scrolled along with the menu rather than just the menu itself.

4.) perform any testing of UI and fix issues arising

**Comment by jrosindell** [comment]
5.) for small screen size get rid of + - up and reset icons.

**Comment by jrosindell** [comment]
7.) Button designs...
i) make partner logo sit underneath main logo on small screen width
ii) experiment with making the OneZoom logo bring up a menu with info, settings and home thus leaving only the compass on the bottom left
iii) address the size icons to make them easier to press especially compass icon

**Comment by lentinj** [comment]
Most of these are safari-specfic, rather than being Mobile problems, e.g.:

* https://benfrain.com/preventing-body-scroll-for-modals-in-ios/
* https://stackoverflow.com/questions/19110052/scrolling-problems-on-a-web-page-with-fixed-header-and-footer-in-ios7

...this makes it slightly tricky to fix, but I've added a bunch of speculative commits above which should reign in a lot of the unwanted scrolling in (1)--(3). If they aren't making a difference then we should probably remove them, particularly https://github.com/OneZoom/OZtree/commit/71774a47e48665055b84404bced0acd88b3c6266 which is adding a useless call on each touchmove if it doesn't achieve anything.

**Comment by lentinj** [comment]
The above tries (6), it's not been fully prettified but I think it works quite well. If we don't like it at all then the commit can be reverted, otherwise we should tweak spacing / size of menu to look better.

**Comment by lentinj** [comment]
@jrosindell any feedback on the stuff here? In particular where it's a Safari issue, I've fixed stuff blind so would be good to know if it's helping or not.

**Comment by jrosindell** [comment]
OK, this one is a bit tricky.... I'm not sure how to solve the problem of testing on Safari apart from to consider arranging another face to face couple of days of coding for the next sprint, unless you have a friend with a spare iPhone to borrow for the day!

Unfortunately there are definitely still issues, not just with the pinging of the menus described above but also with rescaling (via pinch actions) on the menu elements e.g. location menu expanding from the compass icon.  it's very easy with minimal messing around to get the tree into a state where you can't see the search box or other aspects of the underlying tree are misaligned and can't be fixed without reloading the page.  I'm attaching a few screenshots below to show what I mean...

**Comment by jrosindell** [comment]
![img_5533](https://user-images.githubusercontent.com/3009279/41258399-e73561b2-6dc7-11e8-8a72-71134d350804.PNG)
![img_5534](https://user-images.githubusercontent.com/3009279/41258402-eaba29b2-6dc7-11e8-8317-ac355e678914.PNG)
![img_5535](https://user-images.githubusercontent.com/3009279/41258406-ee5a6f78-6dc7-11e8-81d1-db1ea11d050c.PNG)

**Comment by jrosindell** [comment]
![img_5537](https://user-images.githubusercontent.com/3009279/41258469-20598d10-6dc8-11e8-886a-83dae5d1627d.PNG)

**Comment by jrosindell** [comment]
I'd quite like to separate the design questions regarding menus and icons etc. from the bug related issues into separate Git Issues, but I'm not sure if this is what you like?  In any case, for now here are my comments on the layout.

I'm afraid I don't like the single button on the bottom left and the reuse of the icon as a menu.  Yan didn't really like it either when I discussed it with him.  I think that it's still best to have the logo just take you home or to the relevant website and have two icons on the bottom left (maybe the information icon is enough with settings embedded in the menu that pops up).  Given that these are tough decisions and it's possible to sink a lot of time into them and still now be sure we have the right suggestion maybe this is another thing we should host and point Naziha to a page with her mobile (or a few pages with different options) and let her say what she likes / finds intuitive for OTOP at least.

**Comment by jrosindell** [comment]
![img_5536](https://user-images.githubusercontent.com/3009279/41258785-4f6bf826-6dc9-11e8-8ee4-c5877a40a2e9.PNG)

**Comment by jrosindell** [comment]
I'm attaching this to show how it looks on my phone with two logos - I'm not sure I like the way it's wrapping, two logos on top of each other would feel nicer in my view.

**Comment by jrosindell** [comment]
Sorry for being a bit negative here.  I do think it's improving, it's just difficult to fix on an optimal solution.

**Comment by jrosindell** [comment]
Thanks, and agreed on not getting bogged down on the menus - let's save that for a future project.  

The part of the menus that does need addressing for Naziha is just getting a basic solution in place that looks acceptable on mobile.  I think the rearrangements done so far pretty much sort that so we can leave the issue open but not requiring further work just now.

**Comment by jrosindell** [comment]
I saw that the latest hostings on your server for testing reverted to the three button solution in the bottom left - I think that's good to stick with unless something else is requested by Naziha.

**Comment by lentinj** [comment]
Right, I think we're there now. I can't find a way to get iOS to scroll outside the main window now, and the wikipedia iframe still scrolls in both old and new iOS.
